File: solutions/cpp/pangram/1/pangram.cpp
#include "pangram.h"
#include <string>
#include <unordered_set>
#include <cctype>


namespace pangram {
    
    bool is_pangram(std::string input) {
        std::unordered_set<char> letterSet{};
        const size_t UNIQUE_LETTERS{26};
        char lower{'a'};
        
        for (char c : input) {
            if (std::isalpha(c)) {
                lower = static_cast<char>(std::tolower(static_cast<unsigned char>(c)));
                letterSet.insert(lower);
            }
        }

        return letterSet.size() == UNIQUE_LETTERS;    
    }

}  // namespace pangram

File: solutions/go/pangram/1/pangram.go
package pangram

import "unicode"

func IsPangram(input string) bool {
	alphabet := []rune("abcdefghijklmnopqrstuvwxyz")
    runeMap := map[rune]int {}
    for _, r := range alphabet {
        runeMap[r] = 0
    }

    for _, r := range input {
		if !unicode.IsLetter(r) {
            continue
        }
        runeMap[unicode.ToLower(r)]++
    }
    for _, v := range runeMap {
        if v < 1 {
            return false
        }
    }
    return true   
}
